examples/nametoken/src/lib.rs

```diff
@@ -0,0 +1,1719 @@
+    // in the sutuations when one wants to do a partial update not touching
+    // this field, which can be large.
+    data:         StateBox<Vec<u8>, S>,
+}
+
+impl<S: HasStateApi> NameInfo<S> {
+    fn fresh(
+        owner: AccountAddress,
+        name_expires: Timestamp,
+        state_builder: &mut StateBuilder<S>,
+    ) -> Self {
+        NameInfo {
+            owner,
+            name_expires,
+            data: state_builder.new_box(Vec::new()),
+        }
+    }
+}
+
+/// The state for each address.
+#[derive(Serial, DeserialWithState, Deletable, StateClone)]
+#[concordium(state_parameter = "S")]
+struct AddressState<S> {
+    /// The tokens owned by this address.
+    owned_names: StateSet<ContractTokenId, S>,
+    /// The address which are currently enabled as operators for this address.
+    operators:   StateSet<Address, S>,
+}
+
+impl<S: HasStateApi> AddressState<S> {
+    fn empty(state_builder: &mut StateBuilder<S>) -> Self {
+        AddressState {
+            owned_names: state_builder.new_set(),
+            operators:   state_builder.new_set(),
+        }
+    }
+}
+
+/// The contract state.
+// Note: The specification does not specify how to structure the contract state
+// and this could be structured in a more space efficient way depending on the
+// use case.
+#[derive(Serial, DeserialWithState, StateClone)]
+#[concordium(state_parameter = "S")]
+struct State<S: HasStateApi> {
+    /// The state for each account address.
+    state:        StateMap<AccountAddress, AddressState<S>, S>,
+    /// All of the token IDs
+    all_names:    StateMap<ContractTokenId, NameInfo<S>, S>,
+    /// Map with contract addresses providing implementations of additional
+    /// standards.
+    implementors: StateMap<StandardIdentifierOwned, Vec<ContractAddress>, S>,
+}
+
+/// The parameter type for the contract function `setImplementors`. Takes a
+/// standard identifier and list of contract addresses providing
+/// implementations of this standard.
+#[derive(Debug, Serialize, SchemaType)]
+struct SetImplementorsParams {
+    /// The identifier for the standard.
+    id:           StandardIdentifierOwned,
+    /// The addresses of the implementors of the standard.
+    implementors: Vec<ContractAddress>,
+}
+
+#[derive(Debug, Serialize, SchemaType)]
+struct UpdateDataParams {
+    name: String,
+    data: Vec<u8>,
+}
+
+/// The custom errors the contract can produce.
+#[derive(Serialize, Debug, PartialEq, Eq, Reject, SchemaType)]
+enum CustomContractError {
+    /// Failed parsing the parameter.
+    #[from(ParseError)]
+    ParseParams,
+    /// Failed logging: Log is full.
+    LogFull,
+    /// Failed logging: Log is malformed.
+    LogMalformed,
+    /// Failing to register a name because it already exists
+    /// in this contract.
+    NameIsTaken,
+    /// Only accounts can own names
+    OwnerShouldBeAccount,
+    /// The amount does not match the fee
+    IncorrectFee,
+    // some invarians of the state are broken
```

Comment:
Please use `///` instead of `//`. Can you also please use a capital letter for the first letter to be consistent.